lib/http/cookie_jar/abstract_store.rb

```diff
@@ -122,3 +111,16 @@ def cleanup(session = false)
     # self
   end
 end
+
+require 'http/cookie_jar/hash_store'
+
+# Skip loading MozillaStore on JRuby.
+if defined?(JRUBY_VERSION)
+  class HTTP::CookieJar::MozillaStore
+    def initialize(*)
+      raise ArgumentError, "MozillaStore is not supported on JRuby"
```

Comment:
this isn't that of a nice "idiomatic" Ruby approach.
if a constant is not supported it should not be defined (and I would rather prefer `HTTP::CookieJar::MozillaStore` to raise a missing constant, instead of an `ArgumentError` on `initialize`)

Reply:
I agree.  Change this to just `unless defined?(JRUBY_VERSION)` for now and expect someone to come up with an implementation for JRuby.